### ISSUE TITLE
Please use `$$` for comments in DDL instead of `'`

### ISSUE THREAD

**Issue by jdimeo** [hackolade/Snowflake#45]
Currently, in the forward engineered DDL, the normal SQL literal character `'` is used, but that creates invalid SQL if the comment contains a quote or any newlines. Using `$$` avoids this problem.
https://docs.snowflake.com/en/sql-reference/data-types-text.html#dollar-quoted-string-constants

If this won't work, at the very least please escape single quotes.

**Comment by pdesmarets** [comment]
John, please upgrade to v5.2.8 of the application and v0.1.40 of Snowflake.

**Comment by jdimeo** [comment]
Hello @pdesmarets I just installed the latest version of the Snowflake plugin **0.2.7** and it's back to using single quotes for comments

**Comment by pdesmarets** [comment]
John,

It is still present!  But...

The Classic UI is announced as deprecated by Snowflake;  "Snowflake suggests and encourages using Snowsight going forward and that's the reason behind removing the button from Trial accounts, so that users can get familiar with the UI. There is no possibility to restore the Classic UI on accounts in Organizations created after May 30, 2023."

And unfortunately snowsight is not backwards compatible when it comes to character encoding...

We have therefore made it as universal for all of our customer by introducing a choice in the plugin between the old ClassicUI for backward compatibility, and the new Snowsight 
![image](https://github.com/user-attachments/assets/470bc52a-f009-4266-b70f-5cffa82180f1)

And in the CLI, it is documented
--format=< format > where the values for Snowflake are now  ["snowSight", "classicUI"] with default to snowsight

This, we believe, is the best of both worlds to accommodate Snowflake's lack of backward compatibility of Snowsight for character encoding.

Best regards,

Pascal

**Comment by jdimeo** [comment]
Ah thank you! I was able to use `--format` to restore the previous behavior. I'm not at all tracking how Snowsight vs. Classic UI affects forward engineered DDL and the use of `$$`. My comments include apostrophes and Hackolade does not escape them in "Snowsight" mode so I must do manual edits unless I use "Classic UI". I don't _actually_ use Classic UI in Snowflake itself (honestly, I prefer Desktop apps like DBeaver).

**Comment by pdesmarets** [comment]
Hi John,

In our Snowsight forward-engineering we do escape single and double apostrophes:
![image](https://github.com/user-attachments/assets/e77f1697-893f-4bfc-9ca5-73ac6ca69c81)

While in ClassicUI we generate this:
![image](https://github.com/user-attachments/assets/d5786e09-c0b7-49b9-8188-47cf98a60e89)

Can you please help me understand what we need to do differently?

**Comment by jdimeo** [comment]
I misread my error message! Yes, it is being escaped, but my [JOOQ](https://www.jooq.org/) parser didn't understand it.
![image](https://github.com/user-attachments/assets/e54122e3-4d30-4738-b945-159f7564eb58)

My question was more about why this is necessary. The Snowflake engine interprets the DDL regardless of what UI is being used, so why is the UI a factor in the escaping? But I'm unstuck with the `--format` argument so nothing has to change!

**Comment by pdesmarets** [comment]
I don't know anything about JOOQ.  What I do know is that we started to get tickets from users who have no access to the ClassicUI (probably because they are more recent customers than the Snowflake deprecation date) so we needed to find a solution for them while allowing others to keep the previous encoding approach.  Unfortunately this is all due to Snowflake's lack of consistency between their 2 tools.  As a matter of fact they have now suppressed the [Community exchange](https://community.snowflake.com/s/question/0D53r0000BkbYbbCQE/is-it-possible-to-change-coding-in-csv-generated-by-snowsight-with-query-results-in-old-snowflake-interface-exported-csv-has-coding-utf8-bom-but-snowsight-has-only-utf8-its-a-huge-issue-for-countries-like-poland-where-we-have-diacritical-marks-data) that was admitting that much...
